azure-pipelines.yml

```diff
@@ -23,65 +23,59 @@ variables:
   - group: dev-secrets
 
 steps:
-- script: |
-    domain=`echo "$(PBURL)" | awk -F[/:] '{print $4}'`
-    if [[ "$(PBURL)" == *"-dev.buyingcatalogue.digital.nhs.uk"* ]]
-    then
-      ip="51.11.46.27"
-    elif [[ "$(PBURL)" == *"private-test.buyingcatalogue.digital.nhs.uk"* ]]
-    then
-      ip="51.132.54.44"
-    elif [[ "$(PBURL)" == *"public-test.buyingcatalogue.digital.nhs.uk"* ]]
-    then
-      ip="51.132.53.156"
-    fi
-    if [[ -n "$ip" ]]
-    then
-      echo "$ip        $domain" | sudo tee -a /etc/hosts
-      cat domain-template.yaml | sed "s/DOMAIN/$domain/g;s/IP/$ip/g" > domain.yml
-    else
-      touch domain.yml
-    fi
-    curl -k -i $(PBURL) --fail --connect-timeout 30
-    cat domain.yml
-    cat /etc/hosts
-  displayName: 'Resolve domain'
-- task: UseDotNet@2
-  displayName: 'Use DotNet Core 3.1'
+- task: UseGitVersion@5
+  name: gitversion
+  displayName: Work out version
   inputs:
-    version: '3.1.101'
-    includePreviewVersions: false
+    versionSpec: '5.x'
 
-- task: CopyFiles@2
-  enabled: false
-  inputs:
-    Contents: '**'
-    TargetFolder: '$(Build.ArtifactStagingDirectory)'
-    
-- task: PublishBuildArtifacts@1
-  enabled: false
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'drop'
-    publishLocation: 'Container'
-      
-- bash: 
-    "curl -k -i $(PBURL) --fail --connect-timeout 30"
+- bash: |
+    echo "##vso[task.setvariable variable=PBHOSTURL]$(echo "${PBURL#'https://'}")"
+  displayName: Set up PB Host URL
+
+- bash: |
+    echo "51.11.46.27    $PBHOSTURL" | sudo tee -a /etc/hosts
+    curl -k -i $(PBURL) --fail --connect-timeout 30
   displayName: 'Check URL is reachable'
 
-- task: Bash@3
+- bash: |
+    export PBHOSTURL
+    docker-compose stop &&
+    docker-compose rm -f &&
+    docker-compose up --scale chrome=4 -d
+
+    n=0
+    chromeNodeNames=$(docker ps --filter "name=chrome" --format "table {{.Names}}")
+    firstNode=$(echo $chromeNodeNames | cut -d" " -f2)
+
+    until [ "$n" -ge 30 ]
+    do
+      status=$(docker inspect --format "{{json .State.Health.Status }}" "$firstNode" | tr -d '"')
+      
+      if [ "$status" = "healthy" ]; then echo "Selenium grid is ready" && exit 0; fi
+      n=$((n+1)) 
+      sleep 1
+    done
+
+    echo "Selenium grid cannot access $PBHOSTURL"
+    exit 1
```

Comment:
if this works we should put this in a file and run it from there

Reply:
Sure thing, I'll put it in its own script.